x-pack/plugins/ml/public/application/routing/use_active_route.ts

```diff
@@ -30,8 +40,14 @@ export const useActiveRoute = (routesList: MlRoute[]): MlRoute => {
     }
     // Remove trailing slash from the pathname
     const pathnameKey = pathname.replace(/\/$/, '');
-    return routesMap[pathnameKey];
+    return routesMap[pathnameKey] ?? routesMap['/overview'];
   }, [pathname]);
 
-  return activeRoute ?? routesMap['/overview'];
+  useExecutionContext(executionContext, {
+    name: 'Machine Learning',
+    type: 'application',
+    page: activeRoute?.path,
```

Comment:
Just to make sure, does the path contain any unique identifiers (such as a saved object id or anything with free values)?

Reply:
No, `path` in this case is only a path definition, e.g.
- `/explorer`
- `/settings/calendars_list/edit_calendar/:calendarId`
- `/jobs/new_job/recognize`

etc.